include/thread_pool/worker.hpp

```diff
@@ -252,13 +218,13 @@ inline void Worker<Task, Queue>::stop()
 }
 
 template <typename Task, template<typename> class Queue>
-inline void Worker<Task, Queue>::start(const size_t id, WorkerVector& workers, SlottedBag<Queue>& idle_workers, std::atomic<size_t>& num_busy_waiters)
+inline void Worker<Task, Queue>::start(const size_t id, std::shared_ptr<ThreadPoolState<Task, Queue>> state)
 {
     auto expectedState = State::Initialized;
     if (!m_state.compare_exchange_strong(expectedState, State::Running, std::memory_order_acq_rel))
         throw std::runtime_error("Cannot start Worker: it has previously been started or stopped.");
 
-    m_thread = std::thread(&Worker<Task, Queue>::threadFunc, this, id, std::ref(workers), std::ref(idle_workers), std::ref(num_busy_waiters));
+    m_thread = std::thread(&Worker<Task, Queue>::threadFunc, this, id, std::move(state));
```

Comment:
No matter you use move or not, the outcome is the same here because the function argument "state" shared_ptr has another copy outside. And since you are not moving any ownership here it would be more readable if you just copy without a move. Currently when you read the code it seems that you are passing the ownership of the state object to the newly created thread, which is not the case.

Reply:
I thought moving the shared pointer saves the copy construction of the instance passed to std::thread, which saves one atomic ref counter increment (and subsequent decrement)?